### ISSUE TITLE
[DropdownMenu] Sub-menu doesn't handle narrow screen widths very well

### ISSUE THREAD

**Issue by cpmsmith** [radix-ui/primitives#1749]
### Current Behavior

You could argue this is a feature request, but since the side/alignment of sub-menus is strictly managed by Radix, I think that makes this a bug. If there's not enough room right or left to show a submenu, it just overflows.

<img width="307" alt="image" src="https://user-images.githubusercontent.com/9437625/199307917-0684cf68-7656-47aa-9d8b-f7e285f9d50d.png">

### Expected behavior

I think if there's no horizontal room, it should fall back to opening downward or upward. The layout could be a little weird, but it's better than being unreadable.

### Reproducible example

This is observable in the [the default CodeSandbox for DropdownMenu if you enable responsive preview](https://codesandbox.io/s/y9cuiu?module=App.js&resolutionWidth=320&resolutionHeight=675)

### Suggested solution

This can be achieved with the [`fallbackPlacements` option for Floating UI's `flip` modifier](https://github.com/fellowapp/fellow/commit/d85450671cb999eb965073c5d35e3951a564d396).

### Additional context

I know there's also other discussion around exposing more customization of the position logic (which I support). I haven't seen any discussion mentioning the fact that submenus expose even fewer options than other popovers, which should probably be considered in that other discussion.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) | `react-dropdown-menu` | 2.0.1 |
| React            | n/a     | 18.3.0 |
| Browser          | Firefox | 106.0.3 |
| Operating System | macOS | 12.3.1 |

**Comment by cpmsmith** [comment]
>it would likely cause cascading issues with regards to pointer handling.

Yeah, I can certainly see it being complicated, though I suspect not impossible.

>Submenus as they stand aren't really designed to be used on mobile, it's rather a desktop paradigm.

In general you're right, but I do have two quibbles.

1. My primary use case here is a desktop application which supports being used in a narrow window in some cases. Obviously breaking out of the window would be the ideal, but is out of the question.
2. This doesn't necessarily mean Radix has to fall in line or anything, but iOS has been shifting toward using this pattern lately, even including nested menus, such as this one in Reminders, so it's not unheard-of:
   <img src="https://user-images.githubusercontent.com/9437625/199589354-bbc5076a-8adf-43a3-a7f5-72eff9741146.png" width="300" />

**Comment by hichemfantar** [comment]
We can keep the existing desktop behavior for nested menus and for mobile, change it to the way canva handles it.

https://user-images.githubusercontent.com/34947993/210799817-859b2233-b12d-4356-8466-475daceb8ae3.mp4

**Comment by paralin** [comment]
This was raised in #2599 and closed as dupe of this issue.

**Comment by benoitgrelard** [comment]
@paralin there is no need to repost your duplicate issue in this one. GitHub already links to it automatically (look just above your comment). Can you please delete your comment?

**Comment by paralin** [comment]
@benoitgrelard It has an additional example and demonstration of the problem. There is no need to delete.

**Comment by benoitgrelard** [comment]
@paralin The problem described is exactly the same, and your issue and example is still accessible either way.

**Comment by iamJoeTaylor** [comment]
I wrote some CSS like this which seems to work well in most cases.

```
/**
  * Note: this is needed to prevent the popover from being cut off
  * by the viewport when the popover is nested inside another popover
  */
div[data-radix-popper-content-wrapper]
  div[data-radix-popper-content-wrapper]:has(> div[data-side="right"]) {
  left: calc(
    min(
      0px,
      var(--radix-popper-available-width) +
        (-1 * var(--radix-popper-anchor-width))
    )
  ) !important;
}
div[data-radix-popper-content-wrapper]
  div[data-radix-popper-content-wrapper]:has(> div[data-side="left"]) {
  left: calc(
    max(
      0px,
      (-1 * var(--radix-popper-available-width)) +
        var(--radix-popper-anchor-width)
    )
  ) !important;
}
```
